yaas/__main__.py

```diff
@@ -44,38 +32,52 @@ def main():
     parser.add_argument(
         '--raw',
         action='store_true',
-        default=False,
+        default=None,
         help="Print the raw Ambari response.")
 
     parser.add_argument(
         '--debug',
         action='store_true',
-        default=False,
+        default=None,
         help="Print Ambari requests and responses.")
 
-    subparsers = parser.add_subparsers(dest="command")
-
-    commands = {
-        'version' : version,
-        'blueprint': blueprint.command,
-        'cluster': cluster.command,
-        'repo': repo.command,
-        'service': service.command,
-        'task': task.command,
-        'host': host.command,
-        }
-
-    for name, fn in commands.items():
-        subparsers.add_parser(name, help=inspect.getdoc(fn))
-
-    args, extra = parser.parse_known_args(sys.argv[1:])
-    common.args = args
+    subparsers = parser.add_subparsers()
+    commands = [
+        parsers.blueprint,
+        parsers.cluster,
+        parsers.repo,
+        parsers.service,
+        parsers.task,
+        parsers.host,
+        ]
+
+    for command in commands:
+      command.define_subcommand(subparsers)
```

Comment:
Ah, yes! I think that's it... I was leaning `define_subcommands` (plural), but we could totally have a command with no subcommands.